index.js

```diff
@@ -1,127 +1,115 @@
+    this.httpDelegates[path] = delegate;
+  }
+
+  unregisterHTTPDelegate(path) {
+    delete this.httpDelegates[path];
   }
 
-  when(eventConsiguration, handler) {
+  when(eventConfiguration, handler) {
     let handlerWrapper = handler;
     if (!handler.id) {
       handlerWrapper = {
-        "handle": handler,
-        "id": nanoid()
+        'handle': handler,
+        'id': nanoid()
       };
     }
-    if (this.eventIdsToHandlers[eventConsiguration.id]) {
-      this.eventIdsToHandlers[eventConsiguration.id].push(handlerWrapper);
+    if (this.registry.handlers[eventConfiguration.id]) {
+      this.registry.handlers[eventConfiguration.id].push(handlerWrapper);
     }
     else {
-      this.eventIdsToHandlers[eventConsiguration.id] = [handlerWrapper];
+      this.registry.handlers[eventConfiguration.id] = [handlerWrapper];
     }
-    console.log("Registering event " + eventConsiguration.id);
+    console.log('Registering event ' + eventConfiguration.id);
   }
 
-  start(port, callback) {
-    if(port) this.port = port;
-    
-    for (const serviceIndex in this.servicesIdToServices) {
-      let service = this.servicesIdToServices[serviceIndex];
-      if(!service.started){
-        service.start(this);
-        service.started = true;
-      }
-    }
+  start(port, callback = () => { console.log('Reshuffle started!')}) {
+    this.port = port || this.port;
 
-    if (this.shareResources.webserver && !this.webserverStarted){
-      this.shareResources.webserver.listen(this.port ,() => {
-        this.webserverStarted = true;
-     });
+    // Start all services
+    Object.values(this.registry.services).forEach(service => service.start(this))
+
+    // Start the webserver if we have http delegates
+    if(Object.keys(this.httpDelegates).length > 0 && !this.registry.common.webserver) {
+      this.createWebServer();
+      this.registry.common.webserver.listen(this.port ,() => {
+        console.log(`Web server listening on port ${this.port}`)
+      });
     }
-    if (callback) callback();
+
+    callback && callback();
   }
 
   restart(port) {
-    this.start(port, () => { console.log("Refreshing server config")});
+    this.start(port, () => { console.log('Refreshing Reshuffle configuration')});
   }
   
   handleEvent(eventName, event) {
     if (event == null) {
       event = {};
     }
-    event.getService = this.getService.bind(this);
-    let eventHandlers = this.eventIdsToHandlers[eventName];
-    if(eventHandlers.length == 0){
+    
+    const eventHandlers = this.registry.handlers[eventName];
+    if(eventHandlers.length === 0){
       return false;
     }
-    for (let index = 0; index < eventHandlers.length; index++) {
-      const handler = eventHandlers[index];
-      this._p_handle(handler, event);
-    }
+    event.getService = this.getService.bind(this);
+    
+    eventHandlers.forEach(handler => {
+      handler.handle(event);
+    });
+   
     return true;
   }
-  _p_handle(handler, event) {
```

Comment:
this is important to keep it separated for enterprise to overwrite how handle() is managed.

Reply:
okay I'll revert this one, it will make sense to me one day I'm sure :)